SPECS/libatomic_ops/libatomic_ops.spec

```diff
@@ -1,77 +1,90 @@
-Group:      Development/Libraries
-Source0:    http://www.ivmaisoft.com/_bin/atomic_ops/libatomic_ops-%{version}.tar.gz
+Summary:        Atomic memory update operations portable implementation
+Name:           libatomic_ops
+Version:        7.6.12
+Release:        1%{?dist}
+License:        GPLv2 and MIT
 Vendor:         Microsoft Corporation
 Distribution:   Mariner
+Group:          Development/Libraries
+URL:            https://github.com/ivmai/libatomic_ops
+Source0:        https://github.com/ivmai/libatomic_ops/releases/download/v%{version}/%{name}-%{version}.tar.gz
 
 %description
 This package provides semi-portable access to hardware-provided atomic memory update operations on a number of architectures.
 
-%package devel
-Summary:    Development files for the libatomic_ops library
-Group:      Development/Libraries
-Requires:   libatomic_ops
-Provides:   libatomic_ops-devel
-Provides:   libatomic_ops-devel(x86-64)
+%package        devel
+Summary:        Development files for the libatomic_ops library
+Group:          Development/Libraries
+Requires:       %{name} = %{version}-%{release}
 
 %description devel
-Libraries and header files for libatomic_ops library.
+Libraries and header files for the libatomic_ops library.
 
+%package        static
+Summary:        Static libraries for libatomic_ops
+Requires:       %{name}-devel = %{version}-%{release}
+
+%description static
+Static libraries for the libatomic_ops library
 
 %prep
-%setup -q
-%build
-./configure --prefix=%{_prefix}      \
-            --bindir=%{_sbindir}     \
-            --enable-shared \
-            --disable-silent-rules
+%autosetup
 
-make %{?_smp_mflags}
+%build
+%configure \
+    --bindir=%{_sbindir} \
+    --enable-shared \
+    --disable-silent-rules
+%make_build
 
 %install
-make DESTDIR=%{buildroot} install
-find %{buildroot} -name '*.la' -delete
+%make_install
+find %{buildroot} -type f -name "*.la" -delete -print
+# We will package these files manually using the %%license macro
+rm -rf %{buildroot}%{_docdir}/%{name}/{COPYING,LICENSING.txt}
 
 %check
-make check
+%make_build check
 
-%post -p /sbin/ldconfig
-%postun -p /sbin/ldconfig
+%ldconfig_scriptlets
 
 %files
 %defattr(-,root,root)
-%license COPYING
-%{_docdir}/libatomic_ops/COPYING
-%{_docdir}/libatomic_ops/LICENSING.txt
-%{_libdir}/libatomic_ops.so.*
-%{_libdir}/libatomic_ops_gpl.so.*
+%license COPYING doc/LICENSING.txt
+%{_libdir}/libatomic_ops.so.1*
+%{_libdir}/libatomic_ops_gpl.so.1*
 
 %files devel
 %defattr(-,root,root)
 %{_includedir}/*
 %{_docdir}/libatomic_ops/README*
-%{_libdir}/libatomic_ops.a
-%{_libdir}/libatomic_ops_gpl.a
 %{_libdir}/libatomic_ops.so
 %{_libdir}/libatomic_ops_gpl.so
 %{_libdir}/pkgconfig/atomic_ops.pc
 
+%files static
+%defattr(-,root,root)
+%{_libdir}/libatomic_ops.a
+%{_libdir}/libatomic_ops_gpl.a
+
 %changelog
+* Wed Jan 12 2022 Thomas Crain <thcrain@microsoft.com> - 7.6.12-1
+- Upgrade to latest upstream version
+- Move static libraries to a separate subpackage
```

Comment:
Do you think we could just remove them, since we're here? I think only `pulseaudio` explicitly requires the static version and these can often be converted to a dependency on a dynamic library and an additional `Requires: libatomic_ops`.

Reply:
Went ahead and looked at the sources for the packages depending on libatomic_ops. Seems like nothing needs the static libraries, so I went ahead and removed them.